### ISSUE TITLE
Replace Accel-Sim and GPGPU-Sim build system from Make to CMake

### ISSUE THREAD

**Issue by William-An** [accel-sim/accel-sim-framework#279]
The current build system for Accel-Sim and GPGPU-Sim is based on Make, whose output and content are too messy and hard to read. Another issue with the current build system is that extra precautions are needed when adding new modules to the simulator. For instance, several variables like `SIM_OBJ_FILES_DIR` and compiler flags need to be manually passed down the makefiles in each subfolder. By changing to CMake, the weird build dependency bug in #115 can also be addressed.

Switching to CMake will enable the following features:
- Checks for package dependency (`find_package`) and abort config/build if necessary
- Checks for environment variables and abort config/build if necessary
- Cleaner compilation messages and build progress bar support, a sample output looks like the following:
![Image](https://github.com/accel-sim/accel-sim-framework/assets/19645713/69dc917c-ee9d-4a43-8407-da0669c6bab6)
- Automated dependency management 
- Simpler `CMakeLists.txt` compared with `Makefile`
- Switch to different build backends like `ninja` or Visual studio if needed
- Simpler and intuitive build options passing (easily build for release and debug without the need to run setup script)
- Integrate setup script as part of build configuration

**Comment by JRPan** [comment]
Did you implement this already?

**Comment by William-An** [comment]
Nope, but planning to get it to work.

**Comment by JRPan** [comment]
Missing:

- [ ] OpenCL support
- [ ] cuobjdump_to_ptxplus

**Comment by William-An** [comment]
For this to mark complete, what are the remaining tasks? Besides documentation and the above two components compilation?

**Comment by JRPan** [comment]
well I guess you can just write docs and say: 
```
if you want opencl and cuobjdump_to_ptxplus just use make
```

and write up how to use cmake.

**Comment by tgrogers** [comment]
Yeah, at this point, I don't know if anyone cares about OpenCL 1.0 or the
SASS 1.3 ptxplus emulation.
It's not worth maintaining the path for new code, just mention you can use
make if you want to use these older modes.
